### PR TITLE
Fix image_demo bug.

### PR DESCRIPTION
**Pull request by zytx121**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

To handle tensor bboxes in `RotLocalVisualizer`.
reslove #611

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by austinmw** [comment]
@zytx121 Hi, I tried to test this PR, but I seem to get a compatibility issue somehow with https://github.com/open-mmlab/mmrotate/blame/dev-1.x/mmrotate/datasets/transforms/transforms.py#L424

> 
> File "/opt/mmrotate/mmrotate/datasets/transforms/transforms.py", line 427, in transform
>     for instance in results['instances']:
> KeyError: 'instances'

Where the keys in `results` are instead:
`dict_keys(['img_path', 'img_id', 'img', 'img_shape', 'ori_shape', 'scale', 'scale_factor', 'keep_ratio', 'homography_matrix', 'gt_bboxes', 'gt_ignore_flags', 'gt_bboxes_labels', 'gt_masks'])`

File: mmrotate/visualization/local_visualizer.py [hunk 1]
```diff
@@ -68,7 +70,14 @@ def _draw_instances(self, image: np.ndarray, instances: ['InstanceData'],
...
             colors = [bbox_palette[label] for label in labels]
 
-            # convert to qbox
+            if isinstance(bboxes, Tensor):
+                if bboxes.size(-1) == 5:
+                    bboxes = RotatedBoxes(bboxes)
+                elif bboxes.size(-1) == 8:
+                    bboxes = QuadriBoxes(bboxes)
+                else:
+                    NotImplementedError
```
Comment (RangiLyu) [inline mmrotate/visualization/local_visualizer.py:79]: add detailed error message

**Comment by zytx121** [comment]
Hi @austinmw If you want to use `ConvertMask2BoxType`, you need to add `instances` in the meta_keys just like

https://github.com/open-mmlab/mmrotate/blob/766185ed317f99379cb14035a6f9e5cf8a5340ad/configs/_base_/datasets/dota_coco.py#L34

**Review by RangiLyu — APPROVED** [review]

**Comment by austinmw** [comment]
Hi @zytx121, 

I trained an Oriented R-CNN model on SRSDD dataset and exported with mmrotate2torchserve script. I then tried to use the contents of the torchserve export (configs, checkpoint) to run `image_demo.py`, but have run into these bugs. 

Here's my launch training command:

```
mpirun --host algo-1-zwn0q -np 4 --allow-run-as-root --tag-output --oversubscribe -mca btl_tcp_if_include eth0 -mca oob_tcp_if_include eth0 -mca plm_rsh_no_tree_spawn 1 -mca pml ob1 -mca btl ^openib -mca orte_abort_on_non_zero_status 1 -mca btl_vader_single_copy_mechanism none -mca plm_rsh_num_concurrent 1 -x NCCL_SOCKET_IFNAME=eth0 -x NCCL_DEBUG=INFO -x LD_LIBRARY_PATH -x PATH -x SMDATAPARALLEL_USE_SINGLENODE=1 -x FI_PROVIDER=efa -x RDMAV_FORK_SAFE=1 -x LD_PRELOAD=/opt/conda/lib/python3.8/site-packages/gethostname.cpython-38-x86_64-linux-gnu.so -verbose -x NCCL_DEBUG=VERSION smddprun /opt/conda/bin/python3.8 -m mpi4py sagemaker_train.py --cfg-options seed=0 env_cfg.dist_cfg.backend='smddp' val_evaluator.collect_device='gpu' default_hooks.logger.interval=5 default_hooks.visualization.draw=True visualizer.save_dir='/opt/ml/checkpoints' train_dataloader.dataset.data_root='/opt/ml/input/data/srsdd/' val_dataloader.dataset.data_root='/opt/ml/input/data/srsdd/' train_cfg.val_interval=5 train_cfg.max_epochs=20 load_from='https://download.openmmlab.com/mmrotate/v0.1.0/oriented_rcnn/oriented_rcnn_r50_fpn_fp16_1x_dota_le90/oriented_rcnn_r50_fpn_fp16_1x_dota_le90-57c88621.pth' --config configs/oriented_rcnn/oriented-rcnn-le90_r50_fpn_6x_srsdd.py --torchserve True
```

I'm not sure where `ConvertMask2BoxType` comes from can't seem to find it

**Pull request merged**